list2.py

```diff
@@ -44,8 +48,20 @@ def remove_adjacent(nums):
 
 
 def linear_merge(list1, list2):
-    # your code here
-    return
+    len_list1 = len(list1)
+    len_list2 = len(list2)
+    index1, index2 = 0, 0
+    merged_list = []
+
+    while index1 < len_list1 and index2 < len_list2:
```

Comment:
Why not `while index1 < len(list1) and index2 < len(list2)`?

Reply:
That would work as well and be less memory usage, the only reason I did it that way is habit and should of refactored it at the end and did not.